src/Entity/Shipment.php

```diff
@@ -160,6 +169,53 @@ public function setShipmentItems(array $shipment_items) {
   /**
    * {@inheritdoc}
    */
+  public function getShippingRateCarrier() {
```

Comment:
I don't follow why we need to keep track of the shipping method and shipping carrier in the shipment, shouldn't that come from the Rate entity and doesn't need to be stored in it's own fields in the shipment?